BuildConfigGen/Program.cs

```diff
@@ -375,6 +376,20 @@ private static void WriteNodePackageJson(string taskOutputNode, string nodeVersi
             ensureUpdateModeVerifier!.WriteAllText(outputNodePackagePath, nodePackageContent, suppressValidationErrorIfTargetPathDoesntExist: false);
         }
 
+        private static void WriteDevDependencyPackageJson(string taskOutputNode, string nodeVersion)
+        {
+            string outputNodePackagePath = Path.Combine(taskOutputNode, "package.json");
+            JsonNode outputNodePackagePathJsonNode = JsonNode.Parse(ensureUpdateModeVerifier!.FileReadAllText(outputNodePackagePath))!;
+            
+            // Upgrade typescript version for Node 20
+            if (nodeVersion == "^20.3.1") {
```

Comment:
Concern here is this can easily break e.g. if the node version changes than we'd silently stop updating the typescript version.

It would be better to extend the config to add a new property "shouldUpdateTypescript"

(No need to parameterize the typescript version itself -- we can do that when we have multiple typescript versions to support)

Reply:
Added a new property "shouldUpdateTypescript" to the config records.